src/js/controllers/SetupControllers.js

```diff
@@ -416,68 +421,88 @@ angular.module('blocktrail.wallet')
             var secret = randomBytes(32).toString('base64');
             var encryptedSecret = CryptoJS.AES.encrypt(secret, $scope.form.password).toString();
 
-            var postData = {
-                username: $scope.form.username,
-                email: $scope.form.email,
-                password: CryptoJS.SHA512($scope.form.password).toString(),
-                password_score: $scope.form.passwordCheck && $scope.form.passwordCheck.score || 0,
-                platform: $rootScope.isIOS && "iOS" || "Android",
-                version: $rootScope.appVersion,
-                encrypted_secret: encryptedSecret,
-                device_uuid: device.uuid,
-                device_name: ([device.platform, device.model].clean().join(" / ")) || 'Unknown Device',
-                skip_two_factor: true // will make the resulting API key not require 2FA in the future
-            };
-            $http.post(CONFIG.API_URL + "/v1/" + (CONFIG.TESTNET ? "tBTC" : "BTC") + "/mywallet/register", postData)
+            trackingService.getBrowserFingerprint()
                 .then(function(result) {
-                    trackingService.trackEvent(trackingService.EVENTS.REGISTRATION);
-                    return launchService.storeAccountInfo(_.merge({}, {secret: secret, encrypted_secret: encryptedSecret}, result.data)).then(function() {
-                        $scope.setupInfo.password = $scope.form.password;
+                    return {
```

Comment:
Any reason this was moved inside a promise? Could alternatively just set the fingerprint on the postData here